Modules/_asynciomodule.c

```diff
@@ -2026,6 +2026,24 @@ leave_task(asyncio_state *state, PyObject *loop, PyObject *task)
     return res;
 }
 
+static PyObject *
+swap_current_task_lock_held(PyDictObject *current_tasks, PyObject *loop,
+                            Py_hash_t hash, PyObject *task)
+{
+    PyObject *prev_task;
+    if (_PyDict_GetItemRef_KnownHash(current_tasks, loop, hash, &prev_task) < 0) {
```

Comment:
Should this also use a `lock held` variant?

Reply:
I don't think it makes much of a difference, but we can. I'll add `_PyDict_GetItemRef_KnownHash_LockHeld`.

The `lock held` variants are important for the `SetItem` calls so that we don't try to re-lock the dict. But not so much for the `GetItem` calls, because those have fast paths that do not lock the dict.